pgxn/neon/pagestore_smgr.c

```diff
@@ -1377,22 +1381,90 @@ nm_adjust_lsn(XLogRecPtr lsn)
  * Return LSN for requesting pages and number of blocks from page server
  */
 static XLogRecPtr
-neon_get_request_lsn(bool *latest, RelFileNode rnode, ForkNumber forknum, BlockNumber blkno)
```

Comment:
I wonder if it is strictly needed to add separate get_XXX_request_lsn functions for all kind of requests?
Them contains mostly the same code, just calling different GetLastWrittenLsnFor...  functions.
But may be once again use wLsnCacheRequest parameter?